### PR TITLE
docs: additional load balancing documentation

### PR DESCRIPTION
**Pull request by travisgroth**
## Summary

- Add some more context links and examples for the health checking and load balancing options
- Add the load balancing feature set to `UPGRADING`

## Related issues
https://github.com/pomerium/pomerium/pull/1804
https://github.com/pomerium/pomerium/pull/1796

## Checklist

- [x] reference any related issues
- [x] updated docs
- [ ] updated unit tests
- [x] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

File: docs/docs/topics/load-balancing.md [hunk 1]
```diff
@@ -65,11 +85,26 @@ policy:
...
+policy:
+  - from: https://myapp.localhost.pomerium.io
+    to:
+      - http://myapp-srv-1:8080
+      - http://myapp-srv-2:8080
+      - http://myapp-srv-3:8080
+      - http://myapp-srv-4:8080
+      - http://myapp-srv-5:8080
+    lb_policy: LEAST_REQUEST
+    least_request_lb_config:
```
Comment (travisgroth) [inline docs/docs/topics/load-balancing.md:105]: @wasaga please double check me on this

**Review by desimone — COMMENTED** [review]
LGTM; only question is do we always want to point at `latest` in the envoy documentation or pin to a version?

**Comment by travisgroth** [comment]
> LGTM; only question is do we always want to point at `latest` in the envoy documentation or pin to a version?

I think a pin would be ideal, but we have a lot of links out to envoy and we're not usually very far behind.  Maybe not a major issue?

Something like [this](https://www.npmjs.com/package/vuepress-plugin-global-variables) would be helpful so we'd only have to update the version in one place, but I'm not sure if pulling that in is a good idea.  Thoughts?

**Comment by desimone** [comment]
> I think a pin would be ideal, but we have a lot of links out to envoy and we're not usually very far behind. Maybe not a major issue?

Not major at all, and I don't think a concern for this API surface. I know there are a few places where we are on v2 of the API and envoy has removed support for v2 in 1.7

> Something like [this](https://www.npmjs.com/package/vuepress-plugin-global-variables) would be helpful so we'd only have to update the version in one place, but I'm not sure if pulling that in is a good idea. Thoughts?

Probably not bad idea at all. I didn't know this package existed.

**Comment by travisgroth** [comment]
👍  i'll try adding that plugin.

**Review by desimone — APPROVED** [review]

**Comment by travisgroth** [comment]
> 👍 i'll try adding that plugin.

This breaks some of our doc generation scripts and the formatting in a weird way, so I'm going to punt for the moment.

**Pull request merged**